File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/CompactFrameworkSDK/v1.0.5000/Windows CE/Samples/VC#/Pocket PC/RomanLegion/Score.cs
/*=====================================================================
  File:      Score.cs

  
  ---------------------------------------------------------------------
   Copyright (C) Microsoft Corporation.  All rights reserved.

  This source code is intended only as a supplement to Microsoft
  Development Tools and/or on-line documentation.  See these other
  materials for detailed information regarding Microsoft code samples.

  THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY
  KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
  IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
  PARTICULAR PURPOSE.
=====================================================================*/


using System;
using System.Drawing;
using System.Windows.Forms;
using System.Reflection;
using System.Threading;
using System.ComponentModel;

namespace RomanLegion
{
	public class Score : BaseObj
	{
		// Score Font
		Font m_drawFont;
		SolidBrush m_drawBrush;
		int m_Score;
		int m_Level;

		public Score(GAME game) : base(game)
		{
			// Score Font
			m_drawFont = new Font("Arial", 16,System.Drawing.FontStyle.Bold);

			// Font Color
			m_drawBrush = new SolidBrush(Color.BlueViolet);

			// Create rectangle for drawing.
			m_x = 0;
			m_y = 0;
				
			m_xOld = m_x;
			m_yOld = m_y;
			m_Score = 0;
			m_Level=1;
			m_bAlive=true;
			m_cx = 240;
			m_cy = 50;
		}

		public void Process()
		{
		}

		new public void Draw(Graphics gx)
		{
			// Write the Score
			string Score="Score: "+m_Score+" Level: "+m_Level;
			gx.DrawString(Score, m_drawFont, m_drawBrush,5.0F,10.0F);
		}

		// Expose the Score
		public int Value
		{
			get
			{
				return(m_Score);
			}
			set
			{
				m_Score = value;
			}
		}
		// Expose the Level
		public int Level
		{
			get
			{
				return(m_Level);
			}
			set
			{
				m_Level = value;
			}
		}
	}
}

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/CompactFrameworkSDK/v1.0.5000/Windows CE/Samples/VB/Windows CE/RomanLegion/Score.vb
'
'*------------------------------------------------------------------------------
'*  <copyright from='1997' to='2002' company='Microsoft Corporation'>
'*   Copyright (c) Microsoft Corporation. All Rights Reserved.
'*
'*   This source code is intended only as a supplement to Microsoft
'*   Development Tools and/or on-line documentation.  See these other
'*   materials for detailed information regarding Microsoft code samples.
'*
'*   </copyright>
'*-------------------------------------------------------------------------------
'*
'* File: Score.vb
'*
'

Public Class Score
    Inherits BaseObj

    ' Score Font
    Dim m_drawFont As Font
    Dim m_drawBrush As SolidBrush
    Dim m_Score As Integer
    Dim m_Level As Integer

    Public Sub New(ByVal game As GAME)
        MyBase.New(game)

        ' Score Font
        m_drawFont = New Font("Arial", 16, System.Drawing.FontStyle.Bold)

        m_drawBrush = New SolidBrush(Color.BlueViolet)

        ' Create rectangle for drawing.
        m_x = 0
        m_y = 0

        m_xOld = m_x
        m_yOld = m_y
        m_Score = 0
        m_Level = 1
        m_bAlive = True
        m_cx = 240
        m_cy = 50
    End Sub


    Public Sub Process()
    End Sub


    Public Shadows Sub Draw(ByVal gx As Graphics)
        Dim Score As String
        Score = "Score: " + CStr(m_Score) + " Level: " + CStr(m_Level)
        gx.DrawString(Score, m_drawFont, m_drawBrush, 5.0F, 10.0F)
    End Sub


    ' Expose the Score
    Public Property Value() As Integer
        Get
            Return (m_Score)
        End Get
        Set(ByVal Value As Integer)
            m_Score = Value
        End Set
    End Property

    ' Expose the Level
    Public Property Level() As Integer
        Get
            Return (m_Level)
        End Get
        Set(ByVal Value As Integer)
            m_Level = Value
        End Set
    End Property
End Class
